packages/devtools_app/lib/src/memory/flutter/memory_screen.dart

```diff
@@ -56,16 +76,37 @@ class MemoryBodyState extends State<MemoryBody> {
         Expanded(
           child: Split(
             axis: Axis.vertical,
-            firstChild: MemoryChart(memoryController),
-            secondChild: const Text('Memory Panel TBD'),
+            firstChild: MemoryChart(),
+            secondChild: const Text('Memory Panel TBD capacity'),
             initialFirstFraction: 0.25,
           ),
         ),
       ],
     );
   }
 
+  void _updateListeningState() async {
+    await serviceManager.serviceAvailable.future;
+
+    final MemoryController memoryController = Controllers.of(context).memory;
+    if (memoryController.hasStarted) return;
+
+    await memoryController.startTimeline();
+/*
+      pauseButton.disabled = false;
+      resumeButton.disabled = true;
+
+      vmMemorySnapshotButton.disabled = false;
+      resetAccumulatorsButton.disabled = false;
+      gcNowButton.disabled = false;
+
+      memoryChart.disabled = false;
+*/
+  }
+
   Widget _leftsideButtons() {
+    final MemoryController memoryController = Controllers.of(context).memory;
```

Comment:
instead of making this call in every method controller is needed, why not store a _controller var and set it in didChangeDependencies?

Reply:
You can also do
```dart
class MemoryBodyState extends State {
  MemoryController get _controller => Controllers.of(context).memory;
}
```

Reply:
Done.